### PR TITLE
Added TypeScriptFilter

### PR DESCRIPTION
**Pull request by JanTvrdik**

### CODE REVIEW

File: WebLoader/Filter/TypeScriptFilter.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	 * @param  string $code
+	 * @param  \WebLoader\Compiler $compiler
+	 * @param  string $file
+	 * @return string
+	 */
+	public function __invoke($code, \WebLoader\Compiler $compiler, $file = NULL)
+	{
+		if (pathinfo($file, PATHINFO_EXTENSION) === 'ts') {
+			$out = substr_replace($file, 'js', -2);
+			Process::run("{$this->bin} $file --target ES5 --out $out", NULL, NULL, $this->env);
```
Comment (fprochazka) [inline WebLoader/Filter/TypeScriptFilter.php]: I think that (at least) the filename and output filename should be escaped properly

Comment (JanTvrdik) [inline WebLoader/Filter/TypeScriptFilter.php]: I'm unaware of a simple and correct way to escape arguments passed to shell. Would `escapeshellarg` satisfy you?

Comment (fprochazka) [inline WebLoader/Filter/TypeScriptFilter.php]: That's exactly what I was aiming for :)

**Comment by fprochazka** [comment]
Thank you, merged https://github.com/janmarek/WebLoader/commit/7a64ebe3facec073a3472d11e08b277935959650

**Pull request closed without merging**